src/utils.js

```diff
@@ -313,3 +313,28 @@ export const parseQueryParams = queryParams => {
   });
   return obj;
 };
+
+const keyToOmitRegexes = () =>
+    JSON.parse(process.env.LUMIGO_BLACKLIST_REGEX || '[".*pass.*", ".*key.*"]').map(x => new RegExp(x, 'i'));
+
+export const omitKeys = obj => {
+  if (obj instanceof Array) {
+    return obj.map(omitKeys)
+  }
+  if (typeof obj === "string") {
+    try {
+      return omitKeys(JSON.parse(obj));
+    } catch (e) {
+      return obj;
+    }
+  }
+  if (!obj || typeof obj !== "object") {
+    return obj;
+  }
+  return Object.keys(obj).reduce((newObj, key) => {
+    let value = omitKeys(obj[key]);
+    let shouldOmitKey = keyToOmitRegexes().some(regex => regex.test(key));
```

Comment:
You createing the regexes list every time. you can create this only once (if you decide to dont use simple contains )